test/config-luatest/vshard_test.lua

```diff
@@ -0,0 +1,360 @@
+                    name = "instance-001",
+                    uri = "storage:storage@unix/:./instance-001.iproto",
+                    master = true,
+                },
+            },
+            weight = 1,
+        },
+        ["d1f75e70-6883-d7fe-0087-e582c9c67543"] = {
+            master = "auto",
+            replicas = {
+                ["22222222-2222-2222-0022-222222222222"] = {
+                    name = "instance-003",
+                    uri = "storage:storage@unix/:./instance-003.iproto",
+                },
+                ["50367d8e-488b-309b-001a-138a0c516772"] = {
+                    name = "instance-004",
+                    uri = "storage:storage@unix/:./instance-004.iproto"
+                },
+            },
+            weight = 1,
+        },
+    }
+
+    -- Master storages.
+    exec = 'return vshard.storage.internal.current_cfg'
+    res = g.server_1:eval(exec)
+    t.assert_equals(res.sharding, exp)
+
+    exp = {
+        ["11111111-1111-1111-0011-111111111111"] = {
+            master = "auto",
+            replicas = {
+                ["ef10b92d-9ae9-e7bb-004c-89d8fb468341"] = {
+                    name = "instance-002",
+                    uri = "storage:storage@unix/:./instance-002.iproto",
+                },
+                ["ffe08155-a26d-bd7c-0024-00ee6815a41c"] = {
+                    name = "instance-001",
+                    uri = "storage:storage@unix/:./instance-001.iproto",
+                },
+            },
+            weight = 1,
+        },
+        ["d1f75e70-6883-d7fe-0087-e582c9c67543"] = {
+            replicas = {
+                ["22222222-2222-2222-0022-222222222222"] = {
+                    name = "instance-003",
+                    uri = "storage:storage@unix/:./instance-003.iproto",
+                    master = true,
+                },
+                ["50367d8e-488b-309b-001a-138a0c516772"] = {
+                    name = "instance-004",
+                    uri = "storage:storage@unix/:./instance-004.iproto"
+                },
+            },
+            weight = 1,
+        },
+    }
+    res = g.server_3:eval(exec)
+    t.assert_equals(res.sharding, exp)
+
+    -- Check that basic sharding works.
+    exec = [[
+        function put(v)
+            box.space.a:insert({v.id, v.bucket_id})
+            return true
+        end
+
+        function get(id)
+            return box.space.a:get(id)
+        end
+
+        box.schema.func.create('put')
+        box.schema.role.grant('public', 'execute', 'function', 'put')
+        box.schema.func.create('get')
+        box.schema.role.grant('public', 'execute', 'function', 'get')
+        local format = {{'id', 'unsigned'}, {'bucket_id', 'unsigned'}}
+        a = box.schema.space.create('a', {format = format})
+        a:create_index('id', {parts = {'id'}})
+        a:create_index('bucket_id', {parts = {'bucket_id'}, unique = false})
+    ]]
+    g.server_1:eval(exec)
+    g.server_3:eval(exec)
+    fiber.sleep(1)
```

Comment:
Looks like a potentially flaky test. Why 1 sec, but not 2, 3, 4, etc. It seems we should wrap below asserts into `retrying` helper.

Reply:
I've updated the patch following your suggestion. Thanks!